### PR TITLE
Add docstrings for scheduler and resources

### PR DESCRIPTION
**Pull request by meatballs**

### CODE REVIEW

File: src/conference_scheduler/resources.py [hunk 1]
```diff
@@ -3,6 +3,62 @@
...
+        * It will take place on 2016-09-17
+        * There are two rooms - 'Main Hall' and 'Small Room'
+        * The Main Hall can seat 500 people and the Small Room, 50
+        * It is intended to hold two 30 talks in the morning (from 09:30
+          to 10:00 and from 11:00 to 11:30) and two more in the afternoon
+          (from 14:00 to 14:30 and 15:00 to 15:30)
+
+    We would create the following eight objects::
+
+        Slot(
```
Comment (drvinceknight) [inline src/conference_scheduler/resources.py]: We could doctest the doctstrings too? (Just include the `>>>` and `...` and make sure the doctest script also looks at the source files).

Comment (meatballs) [inline src/conference_scheduler/resources.py]: Aggghhhh!! I meant to do that! I've done it in the scheduler examples.

I'll fix it....

File: docs/conf.py [hunk 2]
```diff
@@ -31,6 +31,7 @@
 # extensions coming with Sphinx (named 'sphinx.ext.*') or your custom
 # ones.
 extensions = [
+    'sphinx.ext.doctest',
```
Comment (drvinceknight) [inline docs/conf.py]: Are we also going to have `doctest` scan these things via the usual test runner? Using this will only work when making the documentation but ideally we'd want to make sure the docs are correct before we get to that stage.

Comment (meatballs) [inline docs/conf.py]: Yep - I'm planning to add that in at some point!

Comment (meatballs) [inline docs/conf.py]: for the moment, I'm running it manually but, given that I have to build the .rst manually anyway, it's not much of an issue

Comment (drvinceknight) [inline docs/conf.py]: It might already be happening depending on how we're running our doctests (I'm not too sure anymore).

Comment (drvinceknight) [inline docs/conf.py]: For travis we might want to think about building the docs too? Like we do on axelrod we just make sure that `make html` doesn't fail. That helps pickup syntax errors before rtd would tell us about them on a merged branch. In this case it would also run the doctests on travis.

Comment (meatballs) [inline docs/conf.py]: Yes - I'd also run `make doctest` too, but I think I'd like to do that in a separate PR

Comment (drvinceknight) [inline docs/conf.py]: Ah I had I assumed `make html` ran the doctests when the plugin was there. Cool.

Comment (meatballs) [inline docs/conf.py]: Locally, it doesn't seem to. I've deliberately tried putting errors in and it still builds the docs quite happily without reporting anything. I seem to have to run `make doctest` separately to see any problems

Comment (drvinceknight) [inline docs/conf.py]: Filing that under "things I learnt today" :+1:
